crates/wasi-common/src/sys/windows/hostcalls_impl/fs.rs

```diff
@@ -120,9 +135,22 @@ pub(crate) fn path_open(
 ) -> Result<File> {
     use winx::file::{AccessMode, CreationDisposition, Flags};
 
+    let is_trunc = oflags & wasi::__WASI_OFLAGS_TRUNC != 0;
+
+    if is_trunc {
+        // Windows requires write access for truncation
+        if !write {
+            return Err(Error::ENOTCAPABLE);
+        }
+        // Windows does not support append mode with truncation
```

Comment:
Can you please add some reference?

Reply:
There is no reference to add other than `GENERIC_WRITE` is required to open a file for truncation and therefore you can't *clear* the `FILE_WRITE_DATA` bit (the former will supersede an off `FILE_WRITE_DATA` bit).

The documentation regarding "append only" files on Windows was pretty weak as it is.

Reply:
This is that truncation requires write access. But `dwDesiredAccess` is a bitmask and we could pass a superset of `FILE_WRITE_DATA | FILE_APPEND_DATA`, on the other hand truncation is specified in `dwCreationDisposition` which is a separate parameter.

Reply:
That's correct that truncation is passed as a flag to `dwCreationDisposition`, but `dwDesiredAccess` is then required to have `GENERIC_WRITE` set, which is the conflicting bit with `FILE_WRITE_DATA` unset (the former includes `FILE_WRITE_DATA`).

Reply:
Semantically, I think requests for append-only truncation should be an error condition since conceptually it doesn't make sense ("preserve what's already there, but only after we delete it all").

Reply:
What forces us to have `FILE_WRITE_DATA` unset?

Well, the semantics are: truncate, then ignore all the attempts to seek and append nevertheless.

Reply:
`FILE_APPEND_DATA` without `FILE_WRITE_DATA` must be used for append-only semantics, otherwise `FILE_WRITE_DATA` will grant the ability to write to any file offset.

Reply:
Oh, I understand now. Then probably `ENOTSUP` would be a little better - this is not an invalid value, but it's not supported by the platform. Please add this explanation as a comment too :)